### PR TITLE
vls: use []byte when storing source

### PR DESCRIPTION
**Pull request by nedpals**
extracted from implement-autcompletion branch

### CODE REVIEW

File: vls/text_synchronization.v [hunk 1]
```diff
@@ -59,6 +58,7 @@ fn (mut ls Vls) did_close(id int, params string) {
 }
 
 fn (mut ls Vls) process_file(source string, uri lsp.DocumentUri) {
+	ls.sources[uri.str()] = source.bytes()
```
Comment (danieldaeschle) [inline vls/text_synchronization.v:61]: it's always a conversion... hmm..

Comment (danieldaeschle) [reply vls/text_synchronization.v:61]: `bytes()` `bytestr()`
why not parsing it to bytes() where it's needed?

**Review by danieldaeschle — APPROVED** [review]

File: vls/text_synchronization.v (hunk 1 above)
Comment (nedpals) [reply vls/text_synchronization.v:61]: the parsing stage which mostly happens on the `did(Open/Change)` methods and the `formatting` method are the only ones who consume the source as string. the rest of the methods doesn't really manipulate the contents and they usually loop all the characters/byte of the content, so it make sense to use []byte.

`formatting` method isn't used all the time so there's no problem to use `bytestr()` while the `did(Open/Change)` part uses the original `source` from the argument itself.

another interesting observation is that arrays ([]byte in particular) are faster in freeing data and much better to manage compared to strings. i tried using strings with the same code structure as using `[]byte` and after many key strokes the program would complain that the string was freed twice (bc two content updates have been executed and applied immediately) and crashes immediately. using []byte on the other hand do not occur any of the problems that string has.

in short: []byte is mostly sufficient for most of the methods and that they are less stressful to manage

Comment (danieldaeschle) [reply vls/text_synchronization.v:61]: ok, thanks for the detailed explanation :)
